src/container.c

```diff
@@ -424,23 +424,24 @@ static int container_binding_file(char *src, char *dest)
 
 	if (stat(src, &st) < 0) {
 		if (errno == ENOENT) {
-			fprintf(stdout, "no dns configured\n");
+			fprintf(stdout, "can not find %s\n", src);
 			return 0;
 		}
 
-		fprintf(stderr, "stat %s failed", src);
+		fprintf(stderr, "stat %s failed: %s\n", src, strerror(errno));
 		return -1;
 	}
 
+	// root filesystem may be read only
 	fd = open(dest, O_CREAT| O_WRONLY, 0644);
```

Comment:
open will return EEXIST when pathname already exists and O_CREAT and O_EXCL were used.
```O_EXCL Ensure that this call creates the file: if this flag is specified in conjunction with O_CREAT, and pathname already exists, then open() will fail.```